framework/doc/content/source/problems/SolutionInvalidity.md

```diff
@@ -0,0 +1,41 @@
+
+To declare a solution as "invalid", use the macro in the following code to mark it. The user can provide a short description of the object (optional) and a message to describe the invalidity (+required+):
+
+!listing /test/src/materials/NonsafeMaterial.C  re=\s+if \(_input_diffusivity > _threshold\).*?\}
+
+!alert tip
+It is recommended to have a short description for each object, especially when you want to mark multiple types of invalid solutions within one object.
+
+If any solution invalidity is detected during the solve, a summary table of solution invalid warnings will be generated and reported at the end of each time step. An example of this output is shown below:
+
+```
+Solution Invalid Warnings:
+------------------------------------------------------------------------------------------------------------------------------
+|                 Object                 | Current | Timestep | Total |                        Message                       |
+------------------------------------------------------------------------------------------------------------------------------
+| NonsafeMaterial::Diffusivity check     |      16 |       64 |    64 | The diffusivity is greater than the threshold value! |
+| NonsafeMaterial::Second thing to check |      16 |       64 |    64 | Extra invalid thing!                                 |
+------------------------------------------------------------------------------------------------------------------------------
+```
+
+1. *Object*: shows the moose object name registered for the solution invalid detection and an optional discription
+2. *Current*: shows the number of solution invalid warnings for the latest iteration
+3. *Timestep*: shows the number of solution invalid warnings for one time step
+4. *Total*: shows the total number of soluton invalid warnings for the solve
+5. *Message*: shows the decription of the solution invalidity
+
+This Solution Invalid Warnings table can be silenced by setting [!param](/Problem/FEProblem/allow_invalid_solution) to 'true'. Then the converged solution will be allowed even if there are still solution invalid warnings, but a message will be generated in the end of the calculation as a reminder:
+
+```
+*** Warning ***
+The Solution Invalidity warnings are detected but silenced! Use Problem/allow_invalid_solution=false to activate
+```
+
+The Solution Invalid Warning can also be printed out immediately after it is detected by setting [!param](/Problem/FEProblem/immediately_print_invalid_solution) to `true`.
+
```

Comment:
Add a note about how the number of warnings can map to the number of quadrature points / nodes /elements affected or could be global. The message should provide enough information to the user to take remedial action

Reply:
Comment has been addressed